### ISSUE TITLE
[Bug]: InlineNotification in Dark mode does not use proper text color

### ISSUE THREAD

**Issue by dabrad26** [carbon-design-system/carbon#11538]
### Package

@carbon/react

### Browser

Safari

### Package version

1.4.0

### React version

18.1.0

### Description

When using lowContrast InlineNotifications in g100 the color for text is wrong. It is using the page text color (white) which makes it not readable.
![Screen Shot 2022-06-02 at 10 20 31 AM](https://user-images.githubusercontent.com/8028956/171687744-ef52bc86-26c7-4097-ad9e-2ec0d0691919.png)

### Reproduction/example

https://stackblitz.com/edit/github-kkpgmp?file=src/App.jsx

### Steps to reproduce

Using Dark mode g100 add inline notification with lowContrast.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by janhassel** [comment]
I believe this is due to the notification component using component tokens.

In your example, if I remove the manual `cds--g100` class from `<body>` and import the theme through scss (see below), it is displayed correctly.

```scss
@use '@carbon/react/scss/themes';
@use '@carbon/react/scss/theme';

:root {  
  @include theme.theme(themes.$g100);
}
```

**Comment by dabrad26** [comment]
I use the class cds—g100 to mimick using the Theme component to set the g100 theme as that is the only way I have found to get it to work in the code example site. 

The cds—g100 should be fine as that is setting the css variables that everything uses for getting color. 

We were told early on that Theme is what to use for implementing dark components. And it seems to work fine for everything but this one component in low contrast mode so I feel this is not really any issue with our setup but that one component not using the css variables properly.

**Comment by tay1orjones** [comment]
I'll reopen to investigate. After looking more closely, we do emit component tokens per zone, I'm unsure why the proper value is not provided when the theme mixin is not called. This was the goal of https://github.com/carbon-design-system/carbon/pull/11227
